### ISSUE TITLE
Pass CMake option to dependency when building from source

### ISSUE THREAD

**Issue by ruilvo** [conan-io/cmake-conan#311]
Hey, 

So I have this dependency, [fftw](https://conan.io/center/fftw), and I am using a set of options that isn't part of any pre-built recipe. 
This is the relevant bit of my conanfile.txt
```
[requires]
fftw/3.3.9

[options]
fftw:shared=False
fftw:openmp=True
fftw:threads=True
fftw:combinedthreads=True
fftw:precision=single
```

I'm runing conan from CMake with the script on this repo with the following options:
`conan_cmake_run(CONANFILE conanfile.txt BASIC_SETUP BUILD missing)`

The problem is that the library builds but without any SIMD enabled. In the CMakeLists present on `...fftw\3.3.9\_\_\source\source_subfolder` I can see the following:
```
option (ENABLE_SSE "Compile with SSE instruction set support" OFF)
option (ENABLE_SSE2 "Compile with SSE2 instruction set support" OFF)
option (ENABLE_AVX "Compile with AVX instruction set support" OFF)
option (ENABLE_AVX2 "Compile with AVX2 instruction set support" OFF)
```

But these options are not part of the recipe, which only offers:
```
    options = {"shared": [True, False],
               "fPIC": [True, False],
               "precision": ["double", "single", "longdouble"],
               "openmp": [True, False],
               "threads": [True, False],
               "combinedthreads": [True, False]}
```

So, how can I enable those options? I tried adding `ENV fftw:ENABLE_AVX=ON` to `conan_cmake_run` without success. Any pointers?

**Comment by uilianries** [comment]
Hi @ruilvo !

You have to update the recipe and add that new option. You can do it internally, for your project only, but also, you can open a PR to https://github.com/conan-io/conan-center-index, with your changes.

All options affect the package ID, if you start to inject options which are not listed on the recipe, you will use an inconsistent package, with known package ID, but executing a different behavior.

Regards.

**Comment by ruilvo** [comment]
Thanks for the quick answer @uilianries. How would one go by editing the recipe internally? I assume using a .py version of the conanfile, but how does one inject behaviour to the dependencies?

**Comment by uilianries** [comment]
>  How would one go by editing the recipe internally?

You need to clone https://github.com/conan-io/conan-center-index repository, enter to fftw folder and run `conan create ...`. You will have an internal copy of package and recipe.

>  assume using a .py version of the conanfile, but how does one inject behaviour to the dependencies?

For fftw dependencies you will need to do the same. When I say "injecting", I mean adding `cmake.definition` without an option. So it won't mapped by the package ID.

**Comment by ruilvo** [comment]
Closed by conan-io/conan-center-index#4671
